crates/nu-command/src/random/uuid.rs

```diff
@@ -13,41 +13,185 @@ impl Command for SubCommand {
+                example: "random uuid -v 1",
+                result: None,
+            },
+            Example {
+                description: "Generate a uuid v3 string (namespace with MD5)",
+                example: "random uuid -v 3 -n dns -s example.com",
+                result: None,
+            },
+            Example {
+                description: "Generate a uuid v5 string (namespace with SHA1)",
+                example: "random uuid -v 5 -n dns -s example.com",
+                result: None,
+            },
+            Example {
+                description: "Generate a uuid v7 string (timestamp + random)",
+                example: "random uuid -v 7",
+                result: None,
+            },
+        ]
     }
 }
 
-fn uuid(call: &Call) -> Result<PipelineData, ShellError> {
+fn uuid(
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+) -> Result<PipelineData, ShellError> {
     let span = call.head;
-    let uuid_4 = Uuid::new_v4().hyphenated().to_string();
 
-    Ok(PipelineData::Value(Value::string(uuid_4, span), None))
+    let version: Option<i64> = call.get_flag(engine_state, stack, "version")?;
+    let version = version.unwrap_or(4);
+
+    let uuid_str = match version {
+        1 => {
+            let ts = Timestamp::now(uuid::timestamp::context::NoContext);
+            let node_id = random_mac_address();
+            let uuid = Uuid::new_v1(ts, &node_id);
+            uuid.hyphenated().to_string()
+        }
+        3 => {
+            let (namespace, name) = get_namespace_and_name(engine_state, stack, call, span)?;
+            let uuid = Uuid::new_v3(&namespace, name.as_bytes());
+            uuid.hyphenated().to_string()
+        }
+        4 => {
+            let uuid = Uuid::new_v4();
+            uuid.hyphenated().to_string()
+        }
+        5 => {
+            let (namespace, name) = get_namespace_and_name(engine_state, stack, call, span)?;
+            let uuid = Uuid::new_v5(&namespace, name.as_bytes());
+            uuid.hyphenated().to_string()
+        }
+        7 => {
+            let ts = Timestamp::now(uuid::timestamp::context::NoContext);
+            let uuid = Uuid::new_v7(ts);
+            uuid.hyphenated().to_string()
+        }
+        _ => {
+            return Err(ShellError::GenericError {
+                error: format!(
+                    "Unsupported UUID version: {}. Supported versions are 1, 3, 4, 5, and 7.",
+                    version
+                ),
+                msg: "Invalid UUID version".to_string(),
+                span: Some(span),
+                help: Some("Specify version with -v: 1, 3, 4, 5, or 7".to_string()),
+                inner: Vec::new(),
+            });
+        }
+    };
+
+    Ok(PipelineData::Value(Value::string(uuid_str, span), None))
+}
+
+fn random_mac_address() -> [u8; 6] {
+    let mut mac = [0u8; 6];
+    mac.iter_mut().for_each(|byte| *byte = rand::random());
+    mac[0] |= 0x01;
+    mac
+}
```

Comment:
Is it following the spirit of the `UUIDv1` specification when the MAC address is randomly generated? (even if tagging it by MAC can be bad in other ways)

https://en.wikipedia.org/wiki/Universally_unique_identifier#Versions_1_and_6_(date-time_and_MAC_address)

If it is too much effort to provide a sane implementation, I would say we rather omit it or maybe let the user supply it like for the namespace ones.

Reply:
I have made it such that the user has to pass it for now, but am open to any alternatives that we can consider